### PR TITLE
[#31] [UI] As a user, I can see Point Rating Survey Question

### LINKED ISSUE #31: [UI] As a user, I can see Point Rating Survey Question
## Why

* User need to see survey question and point rating

## Acceptance Criteria

* Show question with order/total number
* Show point rating from 0 to 10
* Show next question button
* Show submit button if last question

## Design

Screen Shot 2022-05-10 at 09 01 11

### PR DESCRIPTION
**Pull request by thiennguyen0196**
- Close #31 

## What happened 👀

- Add string for `Not at all Likely` and `Extremely Likely`
- Create view for point rating view
- Integrate point rating view to questions builder

## Insight 📝

- The list of point rating will be scrollable due to the size of different device's screen size

## Proof Of Work 📹

- Android

- iOS

### CODE REVIEW

File: lib/ui/survey_question/survey_answer_views/point_rating_view.dart [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    if (selectedPoint == null) {
+      return Colors.white54;
+    } else {
+      return selectedPoint >= point ? Colors.white : Colors.white54;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
```
Comment (ducbm051291) [inline lib/ui/survey_question/survey_answer_views/point_rating_view.dart:70]: How about calculating `notAtAllLikely` once here 
```suggestion
  @override
  Widget build(BuildContext context) {
    final notAtAllLikely = selectedPoint != null && selectedPoint! <= maxLength / 2
    return Center(
          child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          SingleChildScrollView(
            scrollDirection: Axis.horizontal,
            child: Container(
              height: 60.0,
              decoration: BoxDecoration(
                border: Border.all(color: Colors.white),
                borderRadius: BorderRadius.circular(Dimensions.radiusNormal),
              ),
              child: Row(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                crossAxisAlignment: CrossAxisAlignment.center,
                children: _pointRatingWidgets,
              ),
            ),
          ),
          const SizedBox(height: Dimensions.paddingNormal),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Text(
                context.localization.notAtAllLikely,
                style:
                    notAtAllLikely
                        ? context.textTheme.displaySmall
                            ?.copyWith(color: Colors.white)
                        : context.textTheme.displaySmall
                            ?.copyWith(color: Colors.white54),
              ),
              Text(
                context.localization.extremelyLikely,
                style:  !notAtAllLikely
                    ? context.textTheme.displaySmall
                        ?.copyWith(color: Colors.white)
                    : context.textTheme.displaySmall
                        ?.copyWith(color: Colors.white54),
              ),
            ],
          )
        ],
      ),
    );
  }
```

**Review by ducbm051291 — APPROVED** [review]
The rest LGTM

File: lib/ui/survey_question/survey_answer_views/point_rating_view.dart (hunk 1 above)
Comment (thiennguyen0196) [reply lib/ui/survey_question/survey_answer_views/point_rating_view.dart:70]: Nice one @ducbm051291 👏

Comment (thiennguyen0196) [reply lib/ui/survey_question/survey_answer_views/point_rating_view.dart:71]: Fixed in 1250224cd7a5a6261d138280bac0ac7a148a3c6a

**Comment by nmint8m** [comment]
@thiennguyen0196 

The UI is overloading in the small devices:

**Review by nmint8m — CHANGES_REQUESTED** [review]
Please help me fix the UI.

**Comment by thiennguyen0196** [comment]
@nmint8m I decided to have the view to be scrollable so that in small devices, we can scroll it. 

I do not want the view to have dynamic width because in some super small devices, the width of each item can be too small and it may overlap the point number, so I decided to keep the width of each item and then have the whole view to be scrollable 😄 

https://github.com/nimblehq/flutter-ic-david-bruce/assets/25218255/7fa01d57-3b32-4d44-afc9-1abd07cad213

**Comment by ducbm051291** [comment]
> @nmint8m I decided to have the view to be scrollable so that in small devices, we can scroll it.
> 
> I do not want the view to have dynamic width because in some super small devices, the width of each item can be too small and it may overlap the point number, so I decided to keep the width of each item and then have the whole view to be scrollable 😄
> 
>  Simulator.Screen.Recording.-.iPhone.SE.3rd.generation.-.2023-05-16.at.11.48.02.mp4

IMO, making this UI scrollable for small devices is better than fixing the width and scale of the font. It may lead to the area being difficult to touch, and the user hardly selects the point they want. I'm fine with this implementation.

**Review by nmint8m — APPROVED** [review]
Look good to me.

**Pull request merged**